simple/dgram_waitset.c

```diff
@@ -84,6 +84,13 @@ static int send_recv()
 	struct fi_cq_entry comp;
 	int ret;
 
+	ret = fi_recv(ep, rx_buf, rx_size + ft_rx_prefix_size(), fi_mr_desc(mr),
```

Comment:
> This is a subtlety that is very easy to overlook. How do the other datagram based tests not fail as well?

`fi_dgram` only has a single message from the client to the server. The initialization occurs faster on the server side because the client side is left posting the receive after the address exchange. The client also performs an `snprintf()`. This gives the server some extra time and it (so far) always has a receive buffer posted ahead of time. 

`fi_dgram_pingpong` had a similar problem with the final incoming `ft_finalize` message. This was solved by pre-posting an extra receive on both sides before starting the test. It also has a couple of other things that prevent races such as calling `ft_sync()` at the start of the pingpong phase and also leading with the client. 

I think one of the issues with the design of `dgram_waitset` is that both sides perform a symmetric send call. The other two dgram tests lead with the client.